javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientMethod.java

```diff
@@ -134,14 +134,15 @@ public final String getClientReference() {
      * Get the comma-separated list of parameter declarations for this ClientMethod.
      */
     public final String getParametersDeclaration() {
-        return getParameters().stream().map(ClientMethodParameter::getDeclaration).collect(Collectors.joining(", "));
+        List<ClientMethodParameter> methodParameters = onlyRequiredParameters ? getMethodRequiredParameters() : getMethodParameters();
+        return methodParameters.stream().map(ClientMethodParameter::getDeclaration).collect(Collectors.joining(", "));
     }
 
     /**
      * Get the comma-separated list of parameter names for this ClientMethod.
      */
     public final String getArgumentList() {
-        return getParameters().stream().map(ClientMethodParameter::getName).collect(Collectors.joining(", "));
+        return getMethodParameters().stream().map(ClientMethodParameter::getName).collect(Collectors.joining(", "));
     }
```

Comment:
I've change this code to enable ordering of parameters (required parameters before non-required parameters).